InclusionAndExclusion/FinUnion.lean

```diff
@@ -7,12 +7,17 @@ import InclusionAndExclusion.Auxiliary
 
 open BigOperators
 
+
+/--X_i , β , A , ⋃ X_i-/
 /-- Given finite number of finite sets, List.FinInter returns their union using an inductive way -/
 def List.FinUnion {α β : Type*} [DecidableEq α] [Fintype β] (A : β → Finset α)(L : List β) : Finset α :=
   match L with
   | [] => ∅
   | x :: xs => A x ∪ (xs.FinUnion A)
 
+lemma mem_union_left_not_right (α : Type) (s t : Finset α) (x: α) : x ∈ s ∪ t  → x ∉ S → x ∈ T :=by
```

Comment:
You do not really need this lemma. $x \in s \cup t$ is equivalent with $x\in s \lor x \in t$. You can use rcases to split the two cases.